### DISCUSSION TITLE
TF doesn't convert image? (Ubuntu 20.04.5 LTS)

### DISCUSSION THREAD

**Discussion by Le0X8** [vladmandic/face-api#162] [Q&A]
I have some problems with this piece of code:
```
...
const faceapi = require('@vladmandic/face-api/dist/face-api.node.js');
...

const tfImage = (buffer) => {
    const decoded = faceapi.tf.node.decodeImage(buffer, 3); // This throws the error
    const result = faceapi.tf.expandDims(decoded, 0);
    decoded.dispose();
    return result;
};
...

const image = tfImage(jpegBuffer);
const detections = await faceapi.detectAllFaces(image)
        .withFaceLandmarks()
        .withFaceExpressions()
        .withFaceDescriptors()
        .withAgeAndGender();

...
```

If I execute this code, I only see:

`2023-03-17 13:19:49.248314: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-03-17 13:19:49.408825: F tensorflow/core/platform/env.cc:391] Check failed: -1 != path_length (-1 vs. -1)
Aborted`

Why?

---
_Linux dd36632 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux_

**Comment by vladmandic** [comment]
first couple of questions:
- which version of `@tensorflow/tfjs-node`?
- did you try using tfjs directly without faceapi? i don't think this is related to faceapi at all
- try using `tf.node.decodeJpeg` instead?
- upload the problematic image?

second, `@tensorflow/tfjs-node` packages actual tensorflow c libraries. and tensorflor 2.x does not even have env.cc (it was split into multiple files). so i'm guessing that this is a broken install of `@tensorflow/tfjs-node` where its somehow using wrong version of underlying tensorflow libraries.
